### ISSUE TITLE
"Vein" ore type causes a crash when generating map chunks near y_max 

### ISSUE THREAD

**Issue by FaceDeer** [minetest/minetest#7364]
##### Issue type
 - Bug report
 
##### Minetest version

```
minetest-0.5.0-cf0bceb-win64
```

##### OS / Hardware

Operating system: Windows 10
CPU: Intel Core i7-3770K

##### Summary

While working on a mod that uses a "vein" ore type to add magma conduits to mapgen, I've discovered that if you try to generate map chunks around the elevation of the ore definition's y_max parameter Minetest will immediately crash with no logged information about the cause of the crash.

##### Steps to reproduce

The issue can be reproduced by starting a Minetest Game with mapgen v7 and the following test mod enabled:

[init.txt](https://github.com/minetest/minetest/files/2025100/init.txt) (rename to init.lua)
[depends.txt](https://github.com/minetest/minetest/files/2025077/depends.txt)

The game should crash immediately upon startup.

If you change the y_max ore parameter to something larger than 0 (for example, 500) the game won't crash.

I've also tried setting the height to 1300 and enabling the floatlands, and I still get a crash when I'm up in the sky around the 1300 altitude so it's not just a sea-level issue.

**Comment by SmallJoker** [comment]
Works for me. Please update your Minetest version to a newer version and check whether the issue still persists.

![screenshot_20180522_174421](https://user-images.githubusercontent.com/1497498/40373900-3335696c-5de8-11e8-8840-618303263998.png)

**Comment by paramat** [comment]
Will test.

**Comment by FaceDeer** [comment]
I just downloaded a fresh copy of build 22ebbe1 for Win 64 from [sfan5's builds](https://minetest.kitsunemimi.pw/builds/) and ran it on a different computer (Intel Core i7-5820K) and I still get the crash immediately on starting (there's just enough time for the terrain immediately around the player to spawn).

I used mapgen v7, Minetest Game, and map seed "15".

**Comment by paramat** [comment]
Vein ore hasn't been used much so it's entirely possible there is a long running bug, or a bug uncovered by one of my recent changes.

**Comment by paramat** [comment]
Here's the definition cleaned up, with unnecessary parameters removed:
```
minetest.register_ore({
	ore_type = "vein",
	ore = "default:lava_source",
	wherein = "default:stone",
	y_max = 0,
	y_min = -31000,
	noise_params = {
		octaves = 4,
		seed = 25391,
		spread = {
			y = 800,
			x = 400,
			z = 400
		},
		persist = 0.5,
		flags = "eased",
		offset = 0,
		scale = 3
	},
	noise_threshold = 0.9,
	random_factor = 0,
})
```
With y_max = 0 it runs for me, but the veins have a broken structure, appearing in endless vertical sheets and slices, noise is certainly broken here:

![screenshot_20180523_212023](https://user-images.githubusercontent.com/3686677/40449289-fbbbc894-5ecf-11e8-837f-ccc3d6bed5b6.png)

With y_max = 31000 the veins have the correct tunnel structure.

**Comment by paramat** [comment]
Oddly, with ore y_max = 0 but player spawning at y = -1000 they are fine all the way up to y = 0.

**Comment by paramat** [comment]
Fixed by PR #7371

**Comment by paramat** [comment]
#7371 merged.
